Week-3/assignment-1.js

```diff
@@ -0,0 +1,45 @@
+console.log('--------------------');
+console.log('This is assignment 1');
+
+function memoize(fn) {
+    console.log(`memoize function called`);
+    const cache=new Map()
+    return function(...args){
+        const key = args.toString();
+        if (cache.has(key)){
+           return cache.get(key)
+        }
+
+        cache.set(key,fn(...args))
+        return cache.get(key)
+    }
+    
+}
+
+function add(...numbers){
+    console.log(`add function called`);
+    let variableSent=[...numbers]
+    sum=0
+    for (i in variableSent){
+        sum=sum+variableSent[i]
+    }
+
+   return sum
+}
+
+function time(fn){
+    console.time()
+    console.log(fn())
+    console.timeEnd();
+}
+
+const memoizeAdd=memoize(add)
+
+time(()=> memoizeAdd(100,100))
+time(()=> memoizeAdd(100))
+time(()=> memoizeAdd(100,200))
+time(()=> memoizeAdd(100,100))
+
+// https://github.com/soumya213/p4-rajesh-soumya213/pull/3
+
+// Why add function called for the 4th function
```

Comment:
@soumya213 this we discussed in the 1:1 session. Can you remove this comment. And please change the commit message, its always "Add files via upload". If you uploading each file, please give commit messages based on week and assignment

Reply:
@RajeshSivanesan removed the comment and changed the PR title